zirgen/compiler/picus/picus.cpp

```diff
@@ -288,6 +289,16 @@ class PicusPrinter {
     }
   }
 
+  void visitOp(zirgen::Zhlt::BackOp back) {
+    auto callee = back.getCallee();
+    size_t distance = back.getDistance().getZExtValue();
+    AnySignal signal = signalize(freshName(), back.getType());
```

Comment:
If distance = 0, there are no constraints on `signal` so Picus doesn't know anything about the result of the back. This is sound but not complete, which is okay, but I suspect Picus won't be able to prove determinism of any real uses of `@0`. This probably calls for another test case, and thinking through what constraints we need to add to ensure it's provably deterministic.